interfaces/builtin/uio_test.go

```diff
@@ -107,7 +110,41 @@ SUBSYSTEM=="uio", KERNEL=="uio0", TAG+="snap_consumer_app"`)
 	c.Assert(spec.Snippets(), testutil.Contains, fmt.Sprintf(`TAG=="snap_consumer_app", SUBSYSTEM!="module", SUBSYSTEM!="subsystem", RUN+="%v/snap-device-helper $env{ACTION} snap_consumer_app $devpath $major:$minor"`, dirs.DistroLibExecDir))
 }
 
-func (s *uioInterfaceSuite) TestAppArmorSpec(c *C) {
+func (s *uioInterfaceSuite) TestAppArmorConnectedPlugIgnoresMissingConfigFile(c *C) {
+	log, restore := logger.MockLogger()
+	defer restore()
+
+	builtin.MockEvalSymlinks(&s.BaseTest, func(path string) (string, error) {
+		c.Assert(path, Matches, "/sys/class/uio/uio[0-1]+/device/config")
+		return "", os.ErrNotExist
+	})
+
+	spec := &apparmor.Specification{}
+	// Simulate two UIO connections.
+	c.Assert(spec.AddConnectedPlug(s.iface, s.plug, s.slotGadget0), IsNil)
+	c.Assert(spec.AddConnectedPlug(s.iface, s.plug, s.slotGadget1), IsNil)
+	c.Assert(spec.SecurityTags(), DeepEquals, []string{"snap.consumer.app"})
+	c.Assert(spec.SnippetForTag("snap.consumer.app"), Equals, ""+
+		"/dev/uio0 rw,\n"+
+		"/dev/uio1 rw,\n"+
+		"/sys/devices/platform/**/uio/uio[0-9]** r,  # common rule for all uio connections")
+
+	c.Assert(log.String(), testutil.Contains, "cannot configure not existing uio device config file /sys/class/uio/uio0/device/config")
+	c.Assert(log.String(), testutil.Contains, "cannot configure not existing uio device config file /sys/class/uio/uio1/device/config")
+}
+
+func (s *uioInterfaceSuite) TestAppArmorConnectedPlug(c *C) {
+	builtin.MockEvalSymlinks(&s.BaseTest, func(path string) (string, error) {
+		c.Assert(path, Matches, "/sys/class/uio/uio[0-1]+/device/config")
+		device := "uio0"
+		if !strings.Contains(path, device) {
+			device = "uio1"
+		}
+		// Not representative of actual resolved symlink
```

Comment:
Welcome to make this path more similar to the real thing, but probably that useful.